### PR TITLE
feat: Support manager config environment variable expansion

### PR DESCRIPTION
**Pull request by mrsillydog**
### Proposed Change
- Move from using standard os and yaml methods for reading and unmarshalling manager configuration files to the opentelemetry confmap library
- Add the envprovider factory to allow for environment variable usage within the manager configuration files.
- Minor cleanup of unused variables and attributes within the modified test file

##### Checklist
- [X] Changes are tested
- [X] CI has passed

### CODE REVIEW

File: opamp/config_test.go [hunk 1]
```diff
@@ -188,15 +188,17 @@ func TestToTLS(t *testing.T) {
 				}
 
 				expectedConfig := tls.Config{
-					MinVersion: tls.VersionTLS12,
+					// MinVersion: tls.VersionTLS12,
```
Comment (jsirianni) [inline opamp/config_test.go:191]: Was there an issue w/ this test locally, commented out temporarily?

**Review by jsirianni — CHANGES_REQUESTED** [review]
Working great. One question in tests.

File: opamp/config_test.go (hunk 1 above)
Comment (mrsillydog) [reply opamp/config_test.go:191]: My IDE simply flagged the sections I've commented out as unused - they're set, but then never referenced again (in this particular test, the only `require.Equal` call just references `expectedConfig.Certificates`, and the rest of `expectedConfig` is ignored). My take was that commenting out the unused aspects was a middle ground between simply removing all the unused sections (which might be valuable in the future, if this test is ever improved to account for them) and letting the scope of this ticket creep to actually improve the test now. Happy to do either of them if your take is different from mine.

Comment (jsirianni) [reply opamp/config_test.go:191]: We synced n this offline :+1:

**Review by jsirianni — APPROVED** [review]
Looks great assuming CI passes :+1:

**Pull request merged**